src/lightning/fabric/strategies/fsdp.py

```diff
@@ -372,20 +385,19 @@ def save_checkpoint(
         from torch.distributed.checkpoint import FileSystemWriter, save_state_dict
         from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
 
-        modules = [module for module in state.values() if isinstance(module, FSDP)]
+        modules = [module for module in state.values() if isinstance(module, Module)]
```

Comment:
Let's keep this check to find the modules that are FSDP wrapped. Our logic can currently only support one FSDP model + several optimizers in one state.

```suggestion
        modules = [module for module in state.values() if _has_fsdp_modules(module)]
```
and define `_has_fsdp_modules` as `any(isinstance(mod, FullyShardedDataParallel) for mod in module.modules())`